appender-log4j2/src/main/java/com/van/logging/log4j2/Log4j2Appender.java

```diff
@@ -101,10 +101,34 @@ private void close() {
         }
 
         // Deregister shutdown hook to avoid unbounded growth of registered hooks
-        Runtime.getRuntime().removeShutdownHook(this.shutdownHook);
+        if (!isShuttingDown()) {
```

Comment:
The test seems like a roundabout way to go about it. How about just having the try/catch as you have here without the `isShuttingDown()` test?
```
  // Deregister shutdown hook to avoid unbounded growth of registered hooks
  try {
    Runtime.getRuntime().removeShutdownHook(this.shutdownHook);
  } catch (IllegalStateException e) {
    // Swallow the exception 
    VansLogger.logger.warn("Already shutting down. Cannot remove shutdown hook.")
  } catch (final Exception e) {
    // Swallow the exception cause
    VansLogger.logger.error("Error while removing shutdown hook", e);
  }

  // Flush and stop the cache associated with this appender
  ...
```

Reply:
Done!